sdk/src/azure/core/az_mqtt5_rpc_client_hfsm.c

```diff
@@ -505,19 +497,16 @@ static az_result ready(az_event_policy* me, az_event event)
               event_data->correlation_id),
           &this_policy->_internal.property_bag);
 
-      if (!_az_span_is_valid(event_data->content_type, 1, false))
+      if (!az_span_is_content_equal(event_data->content_type, AZ_SPAN_EMPTY))
```

Comment:
Also, please add the fact that it can be AZ_SPAN_EMPTY within the `az_mqtt5_rpc_client_invoke_req_event_data` documentation in `az_mqtt5_rpc_client.h`.

Reply:
Nit: `az_mqtt5_rpc_client_invoke_begin`, as a public API, should take all these arguments as parameters instead of the struct to be in-line with existing IoT functions and the codec. /cc @vaavva

Reply:
Done. Added:

  /**
   * @brief The content type of the request. It can be AZ_SPAN_EMPTY.
   */
  az_span content_type;